src/Rules/Methods/MissingMagicSerializationMethodsRule.php

```diff
@@ -0,0 +1,80 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Methods;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\BetterReflection\Reflector\Exception\IdentifierNotFound;
+use PHPStan\Node\InClassNode;
+use PHPStan\Php\PhpVersion;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use Serializable;
+use function sprintf;
+
+/**
+ * @implements Rule<InClassNode>
+ */
+class MissingMagicSerializationMethodsRule implements Rule
+{
+
+	public function __construct(private PhpVersion $phpversion)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return InClassNode::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$classReflection = $node->getClassReflection();
+		if (!$this->phpversion->serializableRequiresMagicMethods()) {
+			return [];
+		}
+		if (!$classReflection->implementsInterface(Serializable::class)) {
+			return [];
+		}
+		if ($classReflection->isAbstract() || $classReflection->isInterface() || $classReflection->isEnum()) {
+			return [];
+		}
+
+		$messages = [];
+
+		try {
+			$nativeMethods = $classReflection->getNativeReflection()->getMethods();
+		} catch (IdentifierNotFound) {
```

Comment:
When is this thrown?

Reply:
good question. I just copied it from ;)

https://github.com/phpstan/phpstan-src/blob/fb08fb530f60d3510f2864b9935d2f75b29d7840/src/Rules/Methods/MissingMethodImplementationRule.php#L36-L40

Reply:
Alright ,better to keep it, there's probably a reason.